react/LocationInput.tsx

```diff
@@ -130,7 +149,8 @@ const LocationInput: React.FC<Props> = ({
           size="large"
           value={inputValue}
           onChange={handleInputChange}
-          onBlur={handleInputBlur}
+          onBlur={formatAndValidate}
+          onKeyDown={formatAndValidate}
```

Comment:
Added this for a11y matters (hitting the enter key). I suppose those `onBlur` and `onKeyDown` cover all scenarios where we need to format and validate before submitting